### ISSUE TITLE
Security vulnerability in indirect dependency - CVE-2022-1996

### ISSUE THREAD

**Issue by pvnn** [kubernetes/client-go#1117]
What happened:
We are using kubernetes/client-go as a dependency in our operator project to talk to a k8s cluster. A security vulnerability was reported during scans of our repository which point to [CVE-2022-1996](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-1996) which is an indirect dependency of client-go via [kube-openapi](https://github.com/kubernetes/kube-openapi).
Recently the dependency in this repo was updated with: https://github.com/kubernetes/client-go/commit/2a9f95519059bcb745428698c59bf4a376c3f383

However there is still a remnant due to usage of older kube-openapi version:
[k8s.io/kube-openapi v0.0.0-20220603121420-31174f50af60](https://github.com/kubernetes/client-go/blob/master/go.mod#L30)

What you expected to happen:
**It would be great if you could update the kube-openapi dependency to the latest one which uses version 3.8.0 of go-restful.**
i.e.:
`k8s.io/kube-openapi v0.0.0-20220621154418-c39d0f63fac8`
We expect that the security vulnerability is addressed by updating all indirect references of the dependency emicklei/go-restful to 3.8.0 (or higher).

How to reproduce it (as minimally and precisely as possible):
By running security scans (for example using WhiteSource/MEND) on the repository code.

Anything else we need to know:
See details of vulnerability and versions at [CVE-2022-1996](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-1996)

TL;DR:
**Update kube-openapi dependency to the latest one which uses version 3.8.0 of go-restful.**
`k8s.io/kube-openapi v0.0.0-20220621154418-c39d0f63fac8`
_This should help remove all references to the version that contains the vulnerability_

**Comment by dims** [comment]
already done https://github.com/kubernetes/kubernetes/pull/110518

**Comment by pvnn** [comment]
@dims: the issue is to update the kube-openapi dependency. This in turn has a transitive dependency to an older version of go-restful.

This seems to be an issue with several scan tools that seem to look at go.sum file instead of go.mod.
Updating to the latest kube-openapi should help get rid of the older transitive dependency altogether.

**Comment by dims** [comment]
@pvnn please talk to your vuln scanner vendor about false positives. we will not be updating older branches (for this reason).

**Comment by TrygviLauritsen** [comment]
go.mod still seems to refer to the vulnerable version k8s.io/kube-openapi v0.0.0-20220603121420-31174f50af60
